### PR TITLE
THREESCALE-9752: Set :exception as the default forgery protection strategy

### PR DESCRIPTION
**Pull request by jlledom**
**What this PR does / why we need it**:

Check the description and comments at [THREESCALE-9752](https://issues.redhat.com/browse/THREESCALE-9752)

**Which issue(s) this PR fixes** 

[THREESCALE-9752](https://issues.redhat.com/browse/THREESCALE-9752)

**Verification steps** 

See the issue description to reproduce the bug.

### CODE REVIEW

File: app/lib/action_controller/request_forgery_protection/exception_and_reset_strategy.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+      def handle_unverified_request
+        @controller.reset_session
+        exception.handle_unverified_request
+      end
+
+      private
+
+      def exception
+        @exception || ActionController::RequestForgeryProtection::ProtectionMethods::Exception.new(@controller)
```
Comment (akostadinov) [inline app/lib/action_controller/request_forgery_protection/exception_and_reset_strategy.rb:18]:
```suggestion
        @exception ||= ActionController::RequestForgeryProtection::ProtectionMethods::Exception.new(@controller)
```

Or was `||` intentional?

File: app/lib/action_controller/request_forgery_protection/exception_and_reset_strategy.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+module ActionController
+  module RequestForgeryProtection
+    class ExceptionAndResetStrategy
+      def initialize(controller)
+        @controller = controller
+      end
+
+      def handle_unverified_request
+        @controller.reset_session
```
Comment (akostadinov) [inline app/lib/action_controller/request_forgery_protection/exception_and_reset_strategy.rb:11]: should we also update the comment on https://github.com/3scale/porta/blob/5c9335b7bf1a347070524dc4250ddf5ac9563d33/app/lib/authenticated_system.rb#L25

**Comment by jlledom** [comment]
> How does the error page presently look? Can you show a screenshot?

In my local environment, like this. Not sure how it looks in production.

image

File: app/lib/action_controller/request_forgery_protection/exception_and_reset_strategy.rb (hunk 2 above)
Comment (jlledom) [reply app/lib/action_controller/request_forgery_protection/exception_and_reset_strategy.rb:11]: Done

**Review by akostadinov — APPROVED** [review]

**Pull request merged**